### PR TITLE
Fix fork-test failures

### PR DESCRIPTION
**Pull request by leomassazza**
After some research I found the issue is that we are getting a timeout on ethers.js while using a fork. 

There's an issue filled in ethers project [Ethers Issue 1862]( https://github.com/ethers-io/ethers.js/issues/1862).

In the meantime I'm adding a workaround that implies pre-fetching the same data ethers will pull from the underlying network so that is already cached and ethers won't see a timeout.

### CODE REVIEW

File: publish/src/commands/deploy/index.js [hunk 1]
```diff
@@ -416,6 +416,7 @@ const deploy = async ({
 		generateSolidity,
 		runStep,
 		useOvm,
+		useFork,
```
Comment (leomassazza) [inline publish/src/commands/deploy/index.js:419]: I need to send the useFork flag to apply the workarond ONLY if we are in a forked network

File: publish/src/commands/deploy/take-debt-snapshot-when-required.js [hunk 2]
```diff
@@ -39,11 +41,38 @@ module.exports = async ({
...
+					method: 'eth_call',
+					params: [
+						{
+							from: contract.signer.address,
+							to: contract.address,
+							data: '0x759076e5',
+						},
+						'latest',
+					],
+					id: contract.provider._nextId,
```
Comment (leomassazza) [inline publish/src/commands/deploy/take-debt-snapshot-when-required.js:62]: I have a doubt here if I that nextId value needs to be updated. The tests work but may be just pure luck.

File: publish/src/commands/deploy/take-debt-snapshot-when-required.js [hunk 3]
```diff
@@ -39,11 +41,38 @@ module.exports = async ({
...
-	const checkSnapshot = async () => {
-		const [cacheInfo, currentDebt] = await Promise.all([
-			DebtCache.cacheInfo(),
-			DebtCache.currentDebt(),
-		]);
+	// This function is a workaround to pre-cache the CurrentDebt data if running in a fork.
+	// The original issue is that, in a forked environment, DebtCache.currentDebt() takes more than 120 secs which is the hardcoded timeout set in ether's web component
+	// So in order to prevent the call to take that long, we previously execute the call to cache the results and then let ethers do their job
+	// NOTE: This function should be removed once https://github.com/ethers-io/ethers.js/issues/1862 is resolved
+	const cacheCurrentDebtResults = async ({ contract, useFork }) => {
```
[the remark this answers is not in the document]
Comment (justinjmoses) [reply publish/src/commands/deploy/take-debt-snapshot-when-required.js:48]: The inline comment is 💯  - thank you!

**Review by justinjmoses — APPROVED** [review]
:shipit:

**Pull request merged**